scripts/tests/format.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+usage() {
+    cat <<EOF
+Formats all the rust code and Cargo.toml files in the repository.
+
+usage: format.sh [options]
+
+Options:
+    -h, --help      Shows this dialogue
+    -c, --check     Check only, exiting with a non-zero exit code if not
+                    formatted correctly
+    -v, --verbose   Use verbose output
+EOF
+}
+
+# Global configuration variables, read by all the formatting functions
+check=""
+verbose=""
+
+check_return_code() {
+    if [ $? -eq 0 ]; then
+        echo "OK"
+    else
+        exit 1
+    fi
+}
+
+cargo_fmt() {
+    cargofmt_check=""
+    cargofmt_verbose=""
+
+    if [[ ${check} = "check" ]]; then
+        cargofmt_check="-- --check"
+    fi
+
+    if [[ ${verbose} = "verbose" ]]; then
+        cargofmt_verbose="--verbose"
+    fi
+
+    echo "Running cargo formatting ..."
+
+    cargo fmt --all ${cargofmt_verbose} ${cargofmt_check}
+
+    check_return_code
+}
+
+taplo_fmt() {
+    taplo_verbose=""
+    if [[ ${verbose} = "verbose" ]]; then
+        taplo_verbose="--verbose"
+    fi
+
+    echo "Running taplo formatting ..."
+
+    if [[ ${check} = "check" ]]; then
+        taplo fmt --check ${taplo_verbose}
+    else
+        taplo fmt ${taplo_verbose}
+    fi
+
+    check_return_code
+}
+
+# install taplo if it isn't already
+has_taplo=$(whereis taplo)
+if [[ ${has_taplo} = "taplo: " ]]; then
+    cargo install taplo-cli 2>/dev/null
+fi
```

Comment:
We should do the same for cargo fmt, i.e. `whereis cargo fmt` and maybe `cargo install rustfmt`

Reply:
Added in https://github.com/zeitgeistpm/zeitgeist/pull/767/commits/39bb03030fb8d85702a1f4d8733a1c6914ce93f4